java/arcs/core/analysis/PredicateUtils.kt

```diff
@@ -15,62 +15,79 @@ import arcs.core.data.InformationFlowLabel
 import arcs.core.data.InformationFlowLabel.Predicate
 import java.util.BitSet
 
-fun BitSet.toString(transform: ((Int) -> String)?): String {
-    if (transform == null) return "$this"
-    var labels = mutableListOf<String>()
-    var nextBit = nextSetBit(0)
-    while (nextBit != -1) {
-        labels.add(transform(nextBit))
-        if (nextBit == Int.MAX_VALUE) break
-        nextBit = nextSetBit(nextBit + 1)
-    }
-    return labels.joinToString(prefix = "{", postfix = "}")
-}
-
 /**
- * Returns a set of [BitSet] representation for the predicate using the given [indices] to
- * determine the index of an [InformationFlowLabel] in the bitset.
+ * Returns the predicate in Disjunctive Normal Form (DNF) as a set of [BitSet] pairs.
+ *
+ * The first [BitSet] in the pair is a mask that determines the set of valid bits in the second
+ * [BitSet]. The [indices] map is used to determine the index of an [InformationFlowLabel] in the
+ * bitset. Suppose that the universe of labels is {A, B, C}. Here are some examples:
+ *
+ *          Predicate : {(mask, bits)}
+ *                  A : {(100, 100)}
+ *             not(A) : {(100, 000)}
+ *            A and B : {(110, 110)}
+ *       A and not(B) : {(110, 100)}
+ *             A or B : {(100, 100), (010, 010)}
+ *        A or not(B) : {(100, 100), (010, 000)}
+ *   (A and B) or (C) : {(110, 110), (001, 001)}
  */
-fun InformationFlowLabel.Predicate.asSetOfBitSets(
+fun InformationFlowLabel.Predicate.asDNF(
     indices: Map<InformationFlowLabel, Int>
-): Set<BitSet> {
+): Set<Pair<BitSet, BitSet>> {
```

Comment:
I think it would be nicer to define a new data class for this result, rather than using a set of pairs. It's a complex concept, and would warrant having nice names (and documentation) for the two variables, rather than just using a pair.

Reply:
Done. Added a data class named `Conjunct`.